datasets/disfl_qa/disfl_qa.py

```diff
@@ -0,0 +1,192 @@
+
+_DESCRIPTION = """\
+Disfl-QA is a targeted dataset for contextual disfluencies in an information seeking setting,
+namely question answering over Wikipedia passages. Disfl-QA builds upon the SQuAD-v2 (Rajpurkar et al., 2018)
+dataset, where each question in the dev set is annotated to add a contextual disfluency using the paragraph as
+a source of distractors.
+
+The final dataset consists of ~12k (disfluent question, answer) pairs. Over 90% of the disfluencies are
+corrections or restarts, making it a much harder test set for disfluency correction. Disfl-QA aims to fill a
+major gap between speech and NLP research community. We hope the dataset can serve as a benchmark dataset for
+testing robustness of models against disfluent inputs.
+
+Our expriments reveal that the state-of-the-art models are brittle when subjected to disfluent inputs from
+Disfl-QA. Detailed experiments and analyses can be found in our paper.
+"""
+
+_HOMEPAGE = "https://github.com/google-research-datasets/disfl-qa"
+
+_LICENSE = "Disfl-QA dataset is licensed under CC BY 4.0"
+
+_URL = "https://raw.githubusercontent.com/google-research-datasets/Disfl-QA/main/"
+
+_URLS_squad_v2 = {
+    "train": "https://rajpurkar.github.io/SQuAD-explorer/dataset/" + "train-v2.0.json",
+    "dev": "https://rajpurkar.github.io/SQuAD-explorer/dataset/" + "dev-v2.0.json",
+}
+
+
+class DisflQA(datasets.GeneratorBasedBuilder):
+    """A Benchmark Dataset for Understanding Disfluencies in Question Answering"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "squad_v2_id": datasets.Value("string"),
+                "original question": datasets.Value("string"),
+                "disfluent question": datasets.Value("string"),
+                "title": datasets.Value("string"),
+                "context": datasets.Value("string"),
+                "answers": datasets.features.Sequence(
+                    {
+                        "text": datasets.Value("string"),
+                        "answer_start": datasets.Value("int32"),
+                    }
+                ),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
```

Comment:
To automate the task formatting you can add this, as in squad or squad_v2 :)

```suggestion
            citation=_CITATION,
            task_templates=[
                QuestionAnsweringExtractive(
                    question_column="disfluent question", context_column="context", answers_column="answers"
                )
            ],
```